ckanext/datastore/logic/action.py

```diff
@@ -393,6 +393,16 @@ def datastore_search(context, data_dict):
     :param sort: comma separated field names with ordering
                  e.g.: "fieldname1, fieldname2 desc"
     :type sort: string
+    :param include_total: True to return total matching record count
+                          (optional, default: true)
+    :type include_total: bool
+    :param records_format: the format for the records return value:
+        'objects' (default) list of {fieldname1: value1, ...} dicts,
+        'lists' list of [value1, value2, ...] lists,
+        'csv' string containing comma-separated values with no header,
```

Comment:
we already have the dump controller that streams CSV with headers. If we want to add filtering to produce custom CSVs I'd rather add it to that controller.